### DISCUSSION TITLE
Set a callback for when createResource data is loaded.

### DISCUSSION THREAD

**Discussion by robodna** [solidjs/solid#2179] [Q&A]
I am using createResource and would like to set a callback for when 'items' is populated.   I am currently using an effect but seems to be the wrong approach:

```
createEffect(() => {
//Check if latest is not null, there are no errors, and 'state' is ready and not pending or other states.
     if (items.latest && !items.error && items.state === 'ready') {
consle.log(items.latest)
     }
 });
```

What is the correct method to detect when items is loaded?

**Comment by atk** [comment]
You control the fetcher. Even if you cannot influence it directly, adding `.then(data => { callback(); return data })` should be possible.
